nextcord/guild.py

```diff
@@ -2966,3 +2996,161 @@ async def change_voice_state(
         ws = self._state._get_websocket(self.id)
         channel_id = channel.id if channel else None
         await ws.voice_state(self.id, channel_id, self_mute, self_deaf)
+
+    async def fetch_scheduled_events(
+        self,
+        *,
+        with_users: bool = False
+    ) -> ScheduledEventIterator:
+        """Retrieves an :class:`.AsyncIterator` that enables receiving scheduled
+        events on this guild
+
+        .. note::
+
+            This method is an API call. For general usage, consider 
+            :attr:`scheduled_events` instead.
+
+        .. versionadded:: 2.0
+
+        Parameters
+        ----------
+        with_users: Optional[:class:`bool`]
+            If the event should be received with :attr:`ScheduledEvent.users`
+
+        Raises
+        ------
+        HTTPException
+            Getting the events failed.
+
+        Yields
+        ------
+        :class:`.ScheduledEvent`
+            The event with users if applicable
+
+        Examples
+        --------
+
+        Usage ::
+
+            async for event in guild.fetch_scheduled_events():
+                print(event.name)
+
+        Flattening into a list ::
+
+            events = await guild.fetch_scheduled_events().flatten()
+            # events is now a list of ScheduledEvent...
+        """
+        return ScheduledEventIterator(self, with_users=with_users)
+
+    def get_scheduled_event(self, event_id: int) -> Optional[ScheduledEvent]:
```

Comment:
Are these ``Snowflake``? I'm not sure but want to ask

Reply:
get functions use ints as they search in the dictionary ``Dict[int, obj]``, for consistency with other get_x methods